### PR TITLE
Terraform changes for Disaster Recovery

### PR DESCRIPTION
**Pull request by bjss-colin**
Currently a draft raised for awareness.

* Updates the AzureRM provider version
* Retrieves the KeyVault secrets, from the primary environment, when the current environment is DR
* Moves KeyVault to a module that is conditionally deployed when the environment is not DR **
* Conditionally deploys a SQL instance and the databases when the environment is not DR

The KeyVault module transition is a destructive change and will require `terraform mv` to be run to move the existing KeyVault state binding to the new name of `module.azurerm_key_vault.keyvault`

### CODE REVIEW

File: terraform/resource_groups.tf [hunk 1]
```diff
@@ -16,23 +16,6 @@ resource "azurerm_resource_group" "storageaccount" {
   }
 }
 
-resource "azurerm_resource_group" "keyvault" {
```
Comment (ghost) [inline terraform/resource_groups.tf:19]: Moved to the KeyVault module main.tf

File: terraform/resource_groups.tf [hunk 2]
```diff
@@ -16,23 +16,6 @@ resource "azurerm_resource_group" "storageaccount" {
...
-  name           = "${var.project}-${var.environment}-rg-keyvault"
-  location       = var.region
-  tags = {
-    environment  = var.environment,
-    architecture = "new"
-  }
-}
-
-resource "azurerm_resource_group" "sql-server" {
```
Comment (ghost) [inline terraform/resource_groups.tf:28]: Moved to the bc_sql_server module main.tf

File: terraform/modules/keyvault/keyvault_access_policy.tf [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    "Purge",
+    "Recover",
+    "Restore",
+    "SetIssuers",
+    "Update"
+
+  ]
+}
+
+# Should we be managing a resource that we don't provision or deploy?
```
Comment (bjss-colin) [inline terraform/modules/keyvault/keyvault_access_policy.tf:115]: Commented this out with a reason. This changes properties on a KeyVault instance that isn't provisioned by this terraform configuration. We should either bring the core KeyVault instance under this tf config or maybe scrap this block.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -31,163 +31,163 @@ jobs:
...
-      $id=docker images --filter "label=dacpac=true" -q | Select-Object -First 1
-      docker create --name dacpaccontainer $id      
-      mkdir $(Build.ArtifactStagingDirectory)/dacpacs
-      docker cp dacpaccontainer:/deploy-db/. $(Build.ArtifactStagingDirectory)/dacpacs
-      docker rm dacpaccontainer
-    displayName: 'Copy DACPACs'
-
-  - publish: $(build.artifactStagingDirectory)/dacpacs
-    artifact: dacpacs
+# - job: dockerBuildAndPush
```
Comment (steves-bjss) [inline azure-pipelines.yml]: Can you re-enable the rest of the build please.

**Review by steves-bjss — CHANGES_REQUESTED** [review]

**Review by steves-bjss — APPROVED** [review]

**Pull request merged**